lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -196,13 +214,11 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
                 dataGetInfoArr[0].cipheredDataKey =
                     cipherBundle.cipheredDataKey;
             }
-            if (dataStoreSize !== undefined) {
-                metadataStoreParams.size = dataStoreSize;
+            if (mdOnlyHeader === 'true') {
+                metadataStoreParams.size = mdOnlySize;
```

Comment:
does `metadataStoreParams.contentMD5` not need to be updated to `md5`?

Reply:
Since the MD5 gets passed along in the previous function (as the `calculatedHash` in this function), it gets handled by line 221 `metadataStoreParams.contentMD5 = calculatedHash;`